chainerx_cc/chainerx/kernels/indexing.h

```diff
@@ -50,4 +50,11 @@ class WhereASSKernel : public Kernel {
     virtual void Call(const Array& condition, Scalar x, Scalar y, const Array& out) = 0;
 };
 
+class CumsumKernel : public Kernel {
```

Comment:
This kernel should be declared in `kernels/reduction.h`.

Reply:
Ok. Will move it to `reduction.h`. I assumed that since the function doesn't reduce the dimensions of the array,  it could be added to indexing.

Reply:
NumPy classifies `cumsum` same routine as `sum` and `prod`.

Reply:
I have used `MakeRollingPermutation` (defined in `indexing.cc`) in the definition of `cumsum`. It wouldn't be appropriate to redefine it in `reduction.cc` right?